test/e2e/utils.go

```diff
@@ -289,6 +290,22 @@ func currentState(namespace string, node string, currentStateYaml *nmstatev1alph
 	}, ReadTimeout, ReadInterval)
 }
 
+func checkCondition(node string, conditionType nmstatev1alpha1.NodeNetworkStateConditionType, conditionStatus corev1.ConditionStatus) {
+	key := types.NamespacedName{Name: node}
+	var state nmstatev1alpha1.NodeNetworkState
+	Eventually(func() corev1.ConditionStatus {
+		state = nodeNetworkState(key)
+		condition := conditions.Condition(&state, conditionType)
+		if condition == nil {
+			return corev1.ConditionUnknown
+		}
+		return condition.Status
+	}, ReadTimeout, ReadInterval).Should(
+		Equal(conditionStatus),
+		fmt.Sprintf("Actual Conditions:\n%s", conditionsToYaml(state.Status.Conditions)),
```

Comment:
Problem with wrapping also the "Should" is that we don't get the test file line but instead of the checkCondition util.go file line, would be nice to have "EventuallyWithOffset"

Reply:
It is not clear to me how to use "EventuallyWithOffset". Also, it is not used in any other place in kubernetes-nmstate. Would you mind changing all checks in a follow-up?

Reply:
But we don't care about file line at error ?

Reply:
We sure do. But it is not clear to me how to me how to use the function you mention.

https://github.com/onsi/gomega/blob/master/gomega_dsl.go#L213

What should I set the offset to?

Reply:
The function does not exists, that's why we cannot wrap the Should inside the util functions, and we have to expose AsyncAssertion.

Reply:
I see, thanks.